app/controllers/sign_up/registrations_controller.rb

```diff
@@ -8,6 +8,8 @@ class RegistrationsController < ApplicationController
     prepend_before_action :disable_account_creation, only: [:new, :create]
 
     def show
+      return redirect_to sign_up_email_path unless session[:sp].present?
```

Comment:
The start page (`#show`) is only relevant to users coming from an SP, so I wanted to make sure non-SP users didn't accidentally ever wind up on it.  This redirects them to the first step of the sign up process (entering your email) instead.